paper.md

```diff
@@ -2,11 +2,88 @@
+    the presence of a Kubernetes cluster or any such cloud based container orchestrator.
+    With Popper, researchers can build and validate workflows easily in almost 
+    any environment of their choice like a local machine, a supercomputing cluster
+    or a Kubernetes based cloud environment. To exemplify the suitability of the tool, 
+    we present three different case studies where we take examples from Machine 
+    Learning and HPC experiments and turn them into Popper workflows.
 ---
 
 # Introduction {#sec:intro}
 
+<!-- there's a problem of reproducibility in science -->
+
+Scientists and researchers often leave experimental artifacts
+like code, datasets, configuration files, etc. on open-access repositories like 
+Zenodo [@_zenodo_], Figshare [@_figshare_] or Github[@github] without proper
+instructions to reproduce after completing with their research. Therefore,
+it results in one wasting a lot of time trying to figure out how to
+reproduce the experiment from the archived artifacts. Hence, making the process 
+extremely inefficient and cumbersome. This problem in known as the Reproducibility 
+crisis or Replication crisis in science [@sep-scientific-reproducibility].
+
+![An end-to-end example of a workflow. On the left we have the 
+`.yml` file that defines the workflow. On the right, a pictorial 
+representation of it.](./figures/casestudy.png){#fig:casestudy}
+
+<!-- this is what people have tried with workflow engines -->
+
+The reproducibilty problem [@rosendaal2017reproducibility] has been attempted several times in the past which led to the development of various tools and frameworks which tried 
+to solve the problem in distinct ways [@peng2011reproducible]. Scientific workflow execution engines has 
+been a predominant solution for handling this problem by organizing the steps in a 
+complex scientific workflow as the nodes of a directed acyclic graph (DAG) and 
+executing them in correct order [@cohen2017scientific][@sciprocauto2009workflow][@albrecht2012makeflow].
+Ideally, workflows should improve the reproducibility of scientific applications 
+by making it easier to share and reuse between scientists. However, scientists 
+often find it difficult to reuse others workflows, which is known as _workflow decay_. [@workflow_decay]
+This is mostly because of the differences in the environment where the workflows are developed and the environment where they are being reproduced [@meng2017facilitating]. Some people have tried to solve this problem using the traditional Virtual machine model where they execute every task inside a separate VM [@gruning2018practical][@howe2012virtual]. People built virtual machine images from their development environment and used for setting up VM's in the Cloud. [@virtandnestedvirt2012]
```

Comment:
Please break long lines into multiple ones so that it makes it easier to comment on distinct statements.

First sentence is great 👍 

For the second one, again, we need to avoid using "some" and instead use a metric. Also, I think this is changing the topic to VM's, but up to this point, it was all about workflows, so we might want to create another paragraph that talks about virtualization in general (VM's and containers).

Third statement, besides replacing "people", we might want to say something about the use of VMs in the scientific domain